### PR TITLE
Feat/page control with different sized dots

### PR DESCRIPTION
**Pull request by M-i-k-e-l**

### CODE REVIEW

File: src/components/pageControl/index.js [hunk 1]
```diff
@@ -25,9 +27,27 @@ function getSizeStyle(size, enlargeActive, index, currentPage) {
...
+     * while mediumSize and smallSize are enabled (and disabled when this is disabled).
+     * When set to true there will be maximum of 7 shown.
+     * Only relevant when numOfPages > 5.
+     */
+    limitShownPages: PropTypes.bool,
+    /**
+     * Size of page indicator to the sides of the regular one.
+     * Only relevant when limitShownPages is in effect.
+     */
+    mediumSize: PropTypes.number,
```
Comment (ethanshar) [inline src/components/pageControl/index.js]: Let's unify the mediumSize and smallSize prop into a single prop that can either accpet
- single number value (and we'll add a default behavior) 
- array of values up to 3 values for the `default', `medium` and `small`

File: src/components/pageControl/index.js [hunk 2]
```diff
@@ -72,38 +93,113 @@ export default class PageControl extends BaseComponent {
...
+    this.state = {
+      numOfPagesShown: Math.min(MAX_SHOWN_PAGES, props.numOfPages),
+      largeIndicatorsOffset: 0,
+      pagesOffset: 0,
+      prevPage: undefined
+    };
+
+    if (this.showLimitedVersion(props)) {
+      const {smallSize, mediumSize, size} = props;
+      if (mediumSize >= size || smallSize >= mediumSize) {
```
Comment (ethanshar) [inline src/components/pageControl/index.js]: I see you have a lot of range comparisons in the code. 
Consider using lodash `_.inRange`

File: src/components/pageControl/index.js [hunk 3]
```diff
@@ -1,17 +1,19 @@
 import _ from 'lodash';
```
Comment (ethanshar) [inline src/components/pageControl/index.js]: The code looks great! very minimal and simple (:

File: demo/src/screens/componentScreens/CarouselScreen.js [hunk 4]
```diff
@@ -1,70 +1,98 @@
 import _ from 'lodash';
```
Comment (ethanshar) [inline demo/src/screens/componentScreens/CarouselScreen.js]: There's a separate example screen for the PageControl. 
I think it worth update it a little to show the new capabilities (maybe something simple with Next and Back buttons)
(another open is to use the Carousel as an example screen for the PageControl)

File: demo/src/screens/componentScreens/CarouselScreen.js [hunk 5]
```diff
@@ -1,70 +1,98 @@
...
       <ScrollView>
-        <Text text30 margin-20>Carousel</Text>
+        <Text text30 margin-20>
+          Carousel
+        </Text>
+        <View row marginH-20 centerV>
+          <Text>Limit number of pages shown in page control:</Text>
+          <Switch value={limitShownPages} onValueChange={this.toggleLimitShownPages} marginL-10/>
+        </View>
+        <Picker
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply demo/src/screens/componentScreens/CarouselScreen.js]: Changed to slider

File: demo/src/screens/componentScreens/CarouselScreen.js [hunk 6]
```diff
@@ -1,70 +1,98 @@
...
+          containerStyle={styles.picker}
+          placeholder={TOP_BAR_PROPS.title}
+          floatingPlaceholder
+          value={numberOfPagesShown}
+          onChange={this.setNumberOfPagesShown}
+          topBarProps={TOP_BAR_PROPS}
+        >
+          {_.map([...Array(11).keys()].splice(5), item => (
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply demo/src/screens/componentScreens/CarouselScreen.js]: Removed (changed to slider)

File: src/components/pageControl/index.js [hunk 7]
```diff
@@ -72,38 +93,113 @@ export default class PageControl extends BaseComponent {
...
+      } else {
+        return null;
+      }
+    });
+  }
+
+  renderSameSizeIndicators() {
+    const {numOfPages, size, enlargeActive} = this.props;
+
+    return _.map([...new Array(numOfPages)], (item, index) => this.renderTouchableOpacity(index, size, enlargeActive));
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: Removed (changed to slider)

File: src/components/pageControl/index.js [hunk 8]
```diff
@@ -72,38 +93,113 @@ export default class PageControl extends BaseComponent {
...
+    } else if (index >= largeIndicatorsOffset && index < largeIndicatorsOffset + NUM_LARGE_INDICATORS) {
+      return size;
+    } else if (index === largeIndicatorsOffset - 1 || index === largeIndicatorsOffset + NUM_LARGE_INDICATORS) {
+      return mediumSize;
+    } else if (index === largeIndicatorsOffset - 2 || index === largeIndicatorsOffset + NUM_LARGE_INDICATORS + 1) {
+      return smallSize;
+    }
+  }
+
+  renderTouchableOpacity(index, size, enlargeActive) {
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: Changed to renderIndicator

File: src/components/pageControl/index.js [hunk 9]
```diff
@@ -72,38 +93,113 @@ export default class PageControl extends BaseComponent {
...
+    const {currentPage, color, inactiveColor, onPagePress, spacing} = this.props;
+    return (
+      <TouchableOpacity
+        disabled={_.isUndefined(onPagePress)}
+        onPress={() => onPagePress && onPagePress(index)}
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: This is old code (refactored to a new method)

File: src/components/pageControl/index.js [hunk 10]
```diff
@@ -72,38 +93,113 @@ export default class PageControl extends BaseComponent {
...
+        key={index}
+        style={[
+          styles.pageIndicator,
+          {marginRight: spacing / 2, marginLeft: spacing / 2},
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: Fixed

File: src/components/pageControl/index.js [hunk 11]
```diff
@@ -1,17 +1,19 @@
...
-import {BaseComponent} from '../../commons';
 import TouchableOpacity from '../touchableOpacity';
 import View from '../view';
 
-function getColorStyle(color, inactiveColor, index, currentPage) {
+const MAX_SHOWN_PAGES = 7;
+const NUM_LARGE_INDICATORS = 3;
+
+function getColorStyle(color, inactiveColor, isCurrentPage) {
   const compColor = color || ThemeManager.primaryColor;
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: Done

File: src/components/pageControl/index.js [hunk 12]
```diff
@@ -72,38 +93,113 @@ export default class PageControl extends BaseComponent {
...
+          getSizeStyle(size, enlargeActive, index, currentPage)
+        ]}
+      />
+    );
+  }
+
+  renderDifferentSizeIndicators() {
+    const {numOfPagesShown, pagesOffset} = this.state;
+
+    return _.map([...new Array(numOfPagesShown)], (item, index) => {
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: Fixed

File: src/components/pageControl/index.js (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: Changed

File: src/components/pageControl/index.js (hunk 2 above)
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: 10x, but it does not seem to be relevant here: _.inRange is not-inclusive

File: src/components/pageControl/index.js (hunk 3 above)
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: 10x :)

File: src/components/pageControl/index.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/pageControl/index.js]: Changed

File: demo/src/screens/componentScreens/CarouselScreen.js (hunk 4 above)
Comment (M-i-k-e-l) [reply demo/src/screens/componentScreens/CarouselScreen.js]: Added

**Pull request merged**